src/client/vid/glimp_sdl.c

```diff
@@ -42,14 +43,68 @@ static int last_position_x = SDL_WINDOWPOS_UNDEFINED;
 static int last_position_y = SDL_WINDOWPOS_UNDEFINED;
 static SDL_Window* window = NULL;
 static qboolean initSuccessful = false;
+static char **displayindices = NULL;
+static int num_displays = 0;
+
+const char**
+GLimp_GetDisplayIndices(void)
+{
+	return (const char**)displayindices;
+}
+
+int
+GLimp_GetWindowDisplayIndex(void)
+{
+	return last_display;
+}
+
+int
+GLimp_GetNumVideoDisplays(void)
+{
+	return num_displays;
+}
+
+static void
+ClearDisplayIndices(void)
+{
+	if ( displayindices )
+	{
+		for ( int i = 0; i < num_displays; i++ )
+			free( displayindices[ i ] );
+
+		free( displayindices );
+		displayindices = NULL;
+	}
+}
+
+static void
+InitDisplayIndices()
+{
+	displayindices = malloc( ( num_displays + 1 ) * sizeof( char* ) );
+	for ( int i = 0; i < num_displays; i++ )
+	{
+		displayindices[ i ] = malloc( 11 * sizeof( char ) ); // There are a maximum of 10 digits in 32 bit int + 1 for the NULL terminator
+		snprintf( displayindices[ i ], 11, "%d", i );
+	}
+
+	// The last entry is NULL to indicate the list of strings ends
+	displayindices[ num_displays ] = 0;
+}
 
 // --------
 
 static qboolean
 CreateSDLWindow(int flags, int w, int h)
 {
+	// Reset display index Cvar if out of bounds
+	if ( vid_displayindex->value < 0 || vid_displayindex->value >= num_displays)
+		Cvar_SetValue( "vid_displayindex", 0 );
+
+	// last_position_x and last_position_y aren't really used...
```

Comment:
`last_position_x` and `last_postion_y` is used save a window position between render backend restarts.  
Before create a window - a window is positioned to center of the default screen. And values changed to real values after create window and before destroy window. And if we only change render backend, window will be recreated on the same screen, that prevent issues with the move window to unexpected place or create to big/small window if screens have different resolution.

As for me better change code to something like and save `vid_displayindex` before window close:
```c
if (last_position_x == SDL_WINDOWPOS_UNDEFINED || last_position_y == SDL_WINDOWPOS_UNDEFINED) {
      last_position_x = last_position_y = SDL_WINDOWPOS_UNDEFINED_DISPLAY( (int)vid_displayindex->value );
}
```

Reply:
Alright, I've updated it to work more like that, though I used SDL_WINDOWPOS_ISUNDEFINED() for clarity as SDL_WINDOWPOS_UNDEFINED is the same thing as SDL_WINDOWPOS_UNDEFINED_DISPLAY(0)

Other than that, I also made it so that the "display index" option in the VIDEO menu will be hidden if the user only has a single display so it's not confusing for people that only have one monitor.